jsonrpc/client.go

```diff
@@ -53,37 +53,75 @@ func (sc *Client) AccountNonce(context.Context, string) (*big.Int, error) {
 	panic("not implemented")
 }
 
+func (sc *Client) BlockNumber(ctx context.Context) (*big.Int, error) {
+	var blockNumber big.Int
+	err := sc.c.CallContext(ctx, &blockNumber, "starknet_blockNumber")
+	if err != nil {
+		return nil, err
+	}
```

Comment:
nit: minimize err scope when possible
```suggestion
	if err := sc.c.CallContext(ctx, &blockNumber, "starknet_blockNumber"); err != nil {
		return nil, err
	}
```